kapitan/refs/cmd_parser.py

```diff
@@ -46,8 +48,12 @@ def ref_write(args, ref_controller):
         for line in sys.stdin:
             data += line
     else:
-        with open(file_name) as fp:
+        mime_type = mimetypes.guess_type(file_name)
+        modifier = "r" if "text" in mime_type else "rb"
+        with open(file_name, modifier) as fp:
             data = fp.read()
+            if 'b' in fp.mode:
+                is_binary = True
```

Comment:
Right now this only works when passing a file to `--write`. Using `-` as the filename and using data from `stdin` currently requires the data to be utf-8 encodable (i.e. some kind of text). I think it might come down to a design decision wether data from stdin should always be treated as text or always as binary.
Alternatively, I could imagine adding a `--binary` flag that will always skip UTF-8 encoding

Reply:
I think defaulting to text and having a --binary flag would cover most cases and give people enough flexibility to use this. Most use cases I can think of would work fine with text so it's a good default.